sdks/java/io/google-cloud-platform/src/main/java/org/apache/beam/sdk/io/gcp/bigquery/BigQueryIO.java

```diff
@@ -600,15 +598,7 @@ public void processElement(ProcessContext c) throws Exception {
         rows =
             tuple
                 .get(filesTag)
-                .apply(
-                    WithKeys.of(
-                        new SerializableFunction<String, String>() {
-                          public String apply(String s) {
-                            return s;
```

Comment:
I don't think this is update compatible -- you've changed a reshuffle on *strings* to a reshuffle on *random integers*.

Reply:
Yes, but this code path is only used for bounded collections or triggered batch loads - i.e. there's no update compatibility concern for users of Beam 2.1.